extensions/amp-facebook-comments/1.0/component.js

```diff
@@ -0,0 +1,110 @@
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {
+  MessageType,
+  ProxyIframeEmbed,
+} from '../../../src/preact/component/3p-frame';
+import {dashToUnderline} from '../../../src/string';
+import {deserializeMessage} from '../../../src/3p-frame-messaging';
+import {forwardRef} from '../../../src/preact/compat';
+import {tryParseJson} from '../../../src/json';
+import {useCallback, useLayoutEffect, useState} from '../../../src/preact';
+
+/** @const {string} */
+const TYPE = 'facebook';
+const FULL_HEIGHT = '100%';
+const MATCHES_MESSAGING_ORIGIN = () => true;
+const DEFAULT_TITLE = 'Facebook comments';
+
+/**
+ * @param {!FacebookCommentsDef.Props} props
+ * @param {{current: (!FacebookCommentsDef.Api|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+function FacebookCommentsWithRef(
+  {
+    colorScheme,
+    href,
+    locale: localeProp,
+    numPosts,
+    onReady,
+    orderBy,
+    requestResize,
+    title = DEFAULT_TITLE,
+    ...rest
+  },
+  ref
+) {
+  const [height, setHeight] = useState(FULL_HEIGHT);
+  const messageHandler = useCallback(
+    (event) => {
+      const data = tryParseJson(event.data) ?? deserializeMessage(event.data);
+      if (data['action'] == 'ready') {
+        if (onReady) {
+          onReady();
+        }
+      }
+      if (data['type'] == MessageType.EMBED_SIZE) {
+        const height = data['height'];
+        if (requestResize) {
+          requestResize(height);
+          setHeight(FULL_HEIGHT);
+        } else {
+          setHeight(height);
+        }
+      }
+    },
+    [requestResize, onReady]
+  );
+
+  const [locale, setLocale] = useState(localeProp);
+  useLayoutEffect(() => {
+    if (localeProp) {
+      setLocale(localeProp);
+      return;
+    }
+    const win = ref?.current?.ownerDocument?.defaultView;
+    if (!win) {
+      return;
+    }
+    setLocale(dashToUnderline(win.navigator.language));
+  }, [localeProp, ref]);
+
+  return (
+    <ProxyIframeEmbed
+      contextOptions={{tagName: 'AMP-FACEBOOK-COMMENTS'}}
+      options={{colorScheme, href, locale, numPosts, orderBy}}
+      ref={ref}
+      title={title}
+      {...rest}
+      /* non-overridable props */
+      // We sandbox all 3P iframes however facebook embeds completely break in
+      // sandbox mode since they need access to document.domain, so we
+      // exclude facebook.
+      excludeSandbox
+      matchesMessagingOrigin={MATCHES_MESSAGING_ORIGIN}
+      messageHandler={messageHandler}
+      type={TYPE}
+      wrapperStyle={{height}}
+    />
+  );
+}
+
+const FacebookComments = forwardRef(FacebookCommentsWithRef);
+FacebookComments.displayName = 'FacebookComments'; // Make findable for tests.
```

Comment:
Do items like this get removed for builds to end-users?

Reply:
Doesn't look like it - ran `amp dist --extensions=amp-facebook-comments` and found `qc.displayName="FacebookComments"` inside the file. I'll file a follow up issue for this, good catch.

Reply:
Filed #33961